File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>arr, int k){
    int n=arr.size();
    int r=0,l=0;
    int oc=0,count=0,ans=0;
    
    while(r<n){
        if(arr[r] %2!=0){//if the num is odd increament oc and r ptr
            oc++;
            count=0;
        }
        while(oc==k) { //if have the num odd numbers we need in the subarray
            count++;
           if(arr[l]%2!=0) oc--;  //if the left ptr was on an odd number 
            l++;
             
        }
        ans+=count;
            r++;
    }
    
      return ans;
    }
};

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        length = len(nums)
        check = defaultdict(int)
        check[0] = 1
        
        for index in range(length):
            if nums[index] % 2 == 0:
                nums[index] = 0
            else:
                nums[index] = 1
        
        count = 0
        sums = 0
        for num in nums:
            sums += num
            if sums - k in check:
                count += check[sums - k]
            check[sums] += 1
            
                
                
        return count